### PR TITLE
feat: use global redis instance

### PR DESCRIPTION
**Pull request by ableuler**
/deploy

### CODE REVIEW

File: helm-chart/renku-gateway/templates/deployment.yaml [hunk 1]
```diff
@@ -129,8 +129,17 @@ spec:
               {{ end }}
             - name: GATEWAY_SERVICE_PREFIX
               value: {{ .Values.gatewayServicePrefix | default "/api/" | quote }}
-            - name: GATEWAY_REDIS_HOST
-              value: {{ include "call-nested" (list . "redis" "redis.fullname") }}-master
+            - name: REDIS_HOST
+              value: {{ .Values.global.redis.host | default (include "call-nested" (list . "redis" "redis.fullname")) }}
```
Comment (lorenzo-cavazzi) [inline helm-chart/renku-gateway/templates/deployment.yaml:133]: I think we should not compose the name here. We are accessing `.Values.global.redis.<something>` for all the other env variables, should we keep it coherent and set the name in `.Values.global.redis.host` ?

**Review by lorenzo-cavazzi — COMMENTED** [review]
Very nice!
Look at the inline comments (just a couple of details)

File: helm-chart/renku-gateway/values.yaml [hunk 2]
```diff
@@ -90,14 +90,15 @@ image:
   auth:
     name: renku/renku-gateway
     repository: renku/renku-gateway
-    tag: 'latest'
+    tag: "0.10.1-n001.h61908be"
```
[the remark this answers is not in the document]
Comment (ableuler) [reply helm-chart/renku-gateway/values.yaml:93]: indeed, thanks!

File: app/__init__.py [hunk 3]
```diff
@@ -92,6 +83,34 @@
...
 
+@app.before_request
+def setup_redis_client():
+    """Set up a redis connection to the master by querying sentinel."""
+
+    if "pytest" not in sys.modules:
+
+        if current_app.config["REDIS_IS_SENTINEL"]:
+            sentinel = Sentinel(
+                [(current_app.config["REDIS_HOST"], 26379)],
```
[the remark this answers is not in the document]
Comment (ableuler) [reply app/__init__.py]: done

File: app/__init__.py [hunk 4]
```diff
@@ -92,6 +83,34 @@
...
+                sentinel_kwargs={"password": current_app.config["REDIS_PASSWORD"]},
+            )
+            host, port = sentinel.discover_master("mymaster")
+            current_app.logger.debug(f"Discovered redis master at {host}:{port}")
+        else:
+            (host, port) = (current_app.config["REDIS_HOST"], 6379)
```
[the remark this answers is not in the document]
Comment (ableuler) [reply app/__init__.py]: done

File: helm-chart/renku-gateway/templates/deployment.yaml (hunk 1 above)
Comment (ableuler) [reply helm-chart/renku-gateway/templates/deployment.yaml]: See the discussion on the renku repo about this. However, I removed the now obsolete ugly "call-nested hack" to make this line a bit simpler.

Comment (ableuler) [reply helm-chart/renku-gateway/templates/deployment.yaml:133]: I stand corrected. The hack for this is still needed: https://github.com/helm/helm/issues/4535

**Review by lorenzo-cavazzi — APPROVED** [review]
This works well! Let's ditch the local redis one by one :grin: 

P.S: I restarted the tests cause the deployment was failing due to this temporary error 

```
Error: could not download https://github.com/codecentric/helm-charts/releases/download/keycloak-9.8.1/keycloak-9.8.1.tgz: failed to fetch https://github.com/codecentric/helm-charts/releases/download/keycloak-9.8.1/keycloak-9.8.1.tgz : 503 Service Unavailable
```

**Pull request merged**